storage/api/src/main/java/feast/storage/api/retrieval/BatchRetriever.java

```diff
@@ -0,0 +1,20 @@
+package feast.storage.api.retrieval;
+
+import feast.serving.ServingAPIProto;
+
+import java.io.Serializable;
+
+/**
+ * Interface for implementing user defined retrieval functionality from
+ * Batch/historical stores.
+ */
+public interface BatchRetriever extends Serializable {
+
+    /**
+     * Get all features corresponding to the provided batch features request.
+     *
+     * @param request {@link ServingAPIProto.GetBatchFeaturesRequest} containing requested features and file containing entity columns.
+     * @return {@link ServingAPIProto.Job} if successful, contains the location of the results, else contains the error to be returned to the user.
+     */
+    ServingAPIProto.Job getFeatures(ServingAPIProto.GetBatchFeaturesRequest request);
```

Comment:
Trying to imagine what this implementation looks like in practice. Where would the functionality like loading of entities into the store be? In the store implementation or at a higher level?

Reply:
It would be at the store level. I'm reluctant to prescribe that method because it really depends on the store.

```
                   main
                  thread
                    +
                    |
                    |
getBatchFeatures()->+
                    |
                 generate
                 job ref
                 (UUID)
                    |
                    |            child
                 update          thread
                 job db             +
                    |               |
                    +-------------->+
                    |               |
                    |           getFeatures()
                    |               |
                    |               |
                    |               |
                    |             update
                    |             job db
                    |               |
getJob()+---------->+               |
                    |               |
                    |               |
               get updated          |
               status from          |
                 job db             |
                    +               |
                    |               |
                    |               |
                    v               v
```

Reply:
> I'm reluctant to prescribe that method because it really depends on the store.

Makes sense.